src/filtertransdialog.cpp

```diff
@@ -1244,9 +1246,19 @@ bool mmFilterTransactionsDialog::mmIsRecordMatches(const Model_Checking::Data &t
     else if (mmIsNumberChecked() && (mmGetNumber().empty() ? !tran.TRANSACTIONNUMBER.empty()
         : tran.TRANSACTIONNUMBER.empty() || !tran.TRANSACTIONNUMBER.Lower().Matches(mmGetNumber().Lower())))
         ok = false;
-    else if (mmIsNotesChecked() && (mmGetNotes().empty() ? !tran.NOTES.empty()
-        : tran.NOTES.empty() || !tran.NOTES.Lower().Matches(mmGetNotes().Lower())))
-        ok = false;
+    else if (mmIsNotesChecked()) {
+        if (mmGetNotes().empty() && !tran.NOTES.empty()) ok = false;
+        else if (!mmGetNotes().empty()) {
+            if (tran.NOTES.empty()) ok = false;
+            else if (mmGetNotes().StartsWith("regex[")) {
+                std::string pattern = mmGetNotes().SubString(6, mmGetNotes().find_last_of(']') - 1).ToStdString();
```

Comment:
As it will allow regex[....    and regex[....].  Perhaps we just use regex:......

Reply:
Changed to regex: